src/components/views/settings/ChangePassword.tsx

```diff
@@ -183,7 +183,16 @@ export default class ChangePassword extends React.Component<IProps, IState> {
                     }
                 },
                 (err) => {
-                    this.props.onError(err);
+                    if (err instanceof Error) {
+                        this.props.onError(err);
+                    } else {
+                        this.props.onError(
+                            new UserFriendlyError("Error while changing password: %(error)s", {
+                                error: String(err),
+                                cause: undefined,
+                            }),
+                        );
+                    }
```

Comment:
[SonarCloud coverage](https://github.com/matrix-org/matrix-react-sdk/pull/10615/checks?check_run_id=12907030461) not happy with these files not tested at all but it's not something I'm going to invest in now.

Will need an exception here